src/main/Makefile.am

```diff
@@ -28,10 +28,11 @@ pcvc4_SOURCES = \
 	driver_unified.cpp
 pcvc4_LDADD = \
 	libmain.a \
-	@builddir@/../parser/libcvc4parser.la \
-	@builddir@/../libcvc4.la \
 	$(READLINE_LIBS) \
-	@builddir@/../lib/libreplacements.la
+	@builddir@/../lib/libreplacements.la \
+	@builddir@/../libcvc4.la \
+	@builddir@/../parser/libcvc4parser.la \
+	@builddir@/../util/libutil.la
```

Comment:
That is a really good synopsis of the situation :-)

There are two solutions: 1) Mark these as CVC4_PUBLIC and mark the header as cvc4_public.h and 2) Mark these as CVC4_PUBLIC and mark the header as cvc4_private_library.h. The library file for libcvc4.so will be the same and export the symbol in both cases. The difference is whether we distribute the header for users of libcvc4 or not. This is not really something we want users to see and use, but it is always a bit iffy having the function sitting in the library, not giving source code access to users while at the same time using it ourselves. (This is the debate that has to happen every time cvc4_private_library.h comes up as a reasonable solution.)